tools/clang/tools/dxcompiler/dxcpdbutils.cpp

```diff
@@ -622,7 +615,9 @@ struct DxcPdbUtils : public IDxcPdbUtils2
                    hlsl::DxilMDHelper::kDxilSourceMainFileNameOldMDName) {
         MDTuple *tup = cast<MDTuple>(node.getOperand(0));
         MDString *str = cast<MDString>(tup->getOperand(0));
-        IFR(Utf8ToBlobWide(str->getString(), &m_MainFileName));
+        std::string normalized = hlsl::NormalizePath(str->getString());
+        m_MainFileName = nullptr;
```

Comment:
You added this nullptr assignment. Should there be a check after Utf8ToblobWide to see if m_MainFileName is still nullptr?

Reply:
Setting it to nullptr is just to release the previous value if it was assigned. It's not a danger that `Utf8ToBlobWide` leaves it as null. Added a comment here to clarify.